src/coreclr/tools/Common/TypeSystem/Interop/IL/Marshaller.Aot.cs

```diff
@@ -1037,4 +1041,72 @@ protected override void TransformNativeToManaged(ILCodeStream codeStream)
             StoreManagedValue(codeStream);
         }
     }
+
+    class VariantMarshaller : Marshaller
+    {
+        protected override void AllocManagedToNative(ILCodeStream codeStream)
+        {
+            LoadNativeAddr(codeStream);
+            codeStream.Emit(ILOpcode.initobj, _ilCodeStreams.Emitter.NewToken(NativeType));
+        }
+
+        protected override void TransformManagedToNative(ILCodeStream codeStream)
+        {
+            ILEmitter emitter = _ilCodeStreams.Emitter;
+
+            LoadManagedValue(codeStream);
+            LoadNativeAddr(codeStream);
+
+            var helper = InteropTypes.GetMarshal(Context).GetKnownMethod("GetNativeVariantForObject", null);
+            codeStream.Emit(ILOpcode.call, emitter.NewToken(helper));
+        }
+
+        protected override void TransformNativeToManaged(ILCodeStream codeStream)
+        {
+            ILEmitter emitter = _ilCodeStreams.Emitter;
+
+            LoadNativeAddr(codeStream);
+
+            var helper = InteropTypes.GetMarshal(Context).GetKnownMethod("GetObjectForNativeVariant", null);
+            codeStream.Emit(ILOpcode.call, emitter.NewToken(helper));
+
+            StoreManagedValue(codeStream);
+        }
+
+        protected override void EmitCleanupManaged(ILCodeStream codeStream)
+        {
+            // Only do cleanup if it is IN
+            if (!In)
+            {
+                return;
+            }
+
+            ILEmitter emitter = _ilCodeStreams.Emitter;
+
+            LoadNativeAddr(codeStream);
+            var helper = Context.GetHelperEntryPoint("InteropHelpers", "CleanupVariant");
+            codeStream.Emit(ILOpcode.call, emitter.NewToken(helper));
+        }
+    }
+
+    class VariantBoolMarshaller : Marshaller
+    {
+        protected override void AllocAndTransformManagedToNative(ILCodeStream codeStream)
+        {
+            LoadManagedValue(codeStream);
```

Comment:
`VARIANT_BOOL` marshalling is a bit more complicated. See https://github.com/dotnet/runtime/blob/aaaaeed8c8d2d01dccd1f7fc536ea27eee75f472/src/coreclr/vm/ilmarshalers.cpp#L195-L221 and https://github.com/dotnet/runtime/blob/aaaaeed8c8d2d01dccd1f7fc536ea27eee75f472/src/coreclr/vm/ilmarshalers.h#L1960-L1987